torchmetrics/functional/regression/tweedie_deviance.py

```diff
@@ -0,0 +1,117 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple
+
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+
+
+def _tweedie_deviance_score_update(
+    preds: Tensor,
+    targets: Tensor,
+) -> Tuple[Tensor, Tensor]:
+    """Updates and returns variables required to compute Deviance Score for the given power. Checks for same shape
+    of input tensors.
+
+    Args:
+        preds: Predicted tensor
+        targets: Ground truth tensor
+    """
+    _check_same_shape(preds, targets)
+    return preds, targets
+
+
+def _tweedie_deviance_score_compute(preds: Tensor, targets: Tensor, power: int = 0) -> Tensor:
+    """Computes Cosine Similarity.
+
+    Args:
+        preds: Predicted tensor
+        targets: Ground truth tensor
+        power:
```

Comment:
please update docstring with the additional scenarios where power is not 0,1,2
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_tweedie_deviance.html#sklearn.metrics.mean_tweedie_deviance